lib/rouge/lexers/igorpro.rb

```diff
@@ -0,0 +1,407 @@
+			RemoveFromGizmo RemoveFromGraph RemoveFromLayout RemoveFromTable
+			RemoveImage RemoveLayoutObjects RemovePath Rename RenameDataFolder
+			RenamePath RenamePICT RenameWindow ReorderImages ReorderTraces
+			ReplaceText ReplaceWave Resample ResumeUpdate Reverse Rotate Save
+			SaveData SaveExperiment SaveGraphCopy SaveNotebook
+			SavePackagePreferences SavePICT SaveTableCopy SetActiveSubwindow SetAxis
+			SetBackground SetDashPattern SetDataFolder SetDimLabel SetDrawEnv
+			SetDrawLayer SetFileFolderInfo SetFormula SetIgorHook SetIgorMenuMode
+			SetIgorOption SetMarquee SetProcessSleep SetRandomSeed SetScale
+			SetVariable SetWaveLock SetWaveTextEncoding SetWindow ShowIgorMenus
+			ShowInfo ShowTools Silent Sleep Slider Smooth SmoothCustom Sort
+			SortColumns SoundInRecord SoundInSet SoundInStartChart SoundInStatus
+			SoundInStopChart SoundLoadWave SoundSaveWave SphericalInterpolate
+			SphericalTriangulate SplitString SplitWave sprintf sscanf Stack
+			StackWindows StatsAngularDistanceTest StatsANOVA1Test StatsANOVA2NRTest
+			StatsANOVA2RMTest StatsANOVA2Test StatsChiTest
+			StatsCircularCorrelationTest StatsCircularMeans StatsCircularMoments
+			StatsCircularTwoSampleTest StatsCochranTest StatsContingencyTable
+			StatsDIPTest StatsDunnettTest StatsFriedmanTest StatsFTest
+			StatsHodgesAjneTest StatsJBTest StatsKDE StatsKendallTauTest StatsKSTest
+			StatsKWTest StatsLinearCorrelationTest StatsLinearRegression
+			StatsMultiCorrelationTest StatsNPMCTest StatsNPNominalSRTest
+			StatsQuantiles StatsRankCorrelationTest StatsResample StatsSample
+			StatsScheffeTest StatsShapiroWilkTest StatsSignTest StatsSRTest
+			StatsTTest StatsTukeyTest StatsVariancesTest StatsWatsonUSquaredTest
+			StatsWatsonWilliamsTest StatsWheelerWatsonTest StatsWilcoxonRankTest
+			StatsWRCorrelationTest StructGet StructPut SumDimension SumSeries
+			TabControl Tag TextBox ThreadGroupPutDF ThreadStart Tile TileWindows
+			TitleBox ToCommandLine ToolsGrid Triangulate3d Unwrap URLRequest
+			ValDisplay WaveClear WaveMeanStdv WaveStats WaveTransform wfprintf
+			)
+
+			hdf5Operation = %w(
+			HDF5CreateFile HDF5OpenFile HDF5CloseFile HDF5CreateGroup HDF5OpenGroup
+			HDF5ListGroup HDF5CloseGroup HDF5ListAttributes HDF5AttributeInfo HDF5DatasetInfo
+			HDF5LoadData HDF5LoadImage HDF5LoadGroup HDF5SaveData HDF5SaveImage HDF5SaveGroup
+			HDF5TypeInfo HDF5CreateLink HDF5UnlinkObject HDF5LibraryInfo
+			HDF5DumpState HDF5Dump HDF5DumpErrors
+			)
+
+			object_name = /[a-zA-Z_][a-zA-Z0-9_]*/
+			whitespace  = /[\s\r]+/
+			noLineBreak = /[ \t]+/
+			operator = %r([\#$~!%^&*+=\|?:<>/-])
+			punctuation = /[{}()\[\],.;]/
+			number_float= /0x[a-f0-9]+/i
+			number_hex  = /\d+\.\d+(e[\+\-]?\d+)?/
+			number_int  = /[\d]+(?:_\d+)*/
+
+			state :root do
+				rule %r(//), Comment, :comments
+				mixin :preprocessor
+				mixin :function
+				mixin :variables
+				rule /\b(#{keywords.join('|')})\b/i, Keyword
+				rule /\b(#{igorConstants.join('|')})\b/i, Keyword::Constant
```

Comment:
Please refactor this bit to use a single "identifier" match that tests for inclusion in the keywords/constants/builtins list. You can look at line 178 in the javascript lexer for an example.